components/advanced_create_comment/advanced_create_comment.tsx

```diff
@@ -320,6 +331,81 @@ class AdvancedCreateComment extends React.PureComponent<Props, State> {
         }
     }
 
+    getSelectionText = () => {
+        if (this.state.showQuoteButton) {
+            this.setState({showQuoteButton: false});
+        }
+        const selectionData = Utils.getSelectionData();
+        if (!selectionData) {
+            return;
+        }
+        const {startingSelectedElement, endingSelectedElement, selection, rects, spaceForMultilineSelection} = selectionData;
+
+        if (
+            !startingSelectedElement ||
+            !endingSelectedElement ||
+            startingSelectedElement.id !== endingSelectedElement.id ||
+            !startingSelectedElement.firstElementChild?.id.includes('rhsPostMessageText') ||
+            !endingSelectedElement.firstElementChild?.id.includes('rhsPostMessageText')
+        ) {
+            return;
+        }
+
+        const text = selection.toString();
+        if (text === '') {
+            return;
+        }
+        const quoteButtonPosition = selection.anchorOffset < selection.focusOffset ? 'bottom' : 'top';
+        const {positionX, positionY} = this.getQuoteButtonCoords(
+            quoteButtonPosition,
+            rects,
+            spaceForMultilineSelection,
+            startingSelectedElement,
+        );
+        this.setState({
+            mousePositionX: positionX + 'px',
+            mousePositionY: positionY + 'px',
+            showQuoteButton: true,
+            quoteText: text,
+            quoteButtonPosition,
+        });
+    }
+
+    handlePostQuote = () => {
+        const currentMessage = this.state.draft?.message === '' ? '' : `${this.state.draft?.message}\n`;
+        this.applyMarkdown({
+            message: `${currentMessage} > ${this.state.quoteText}\n\n`,
+            markdownMode: 'bold',
+            selectionStart: null,
+            selectionEnd: null,
+        });
+        this.setState({
+            showQuoteButton: false,
+            quoteText: '',
+            mousePositionX: undefined,
+            mousePositionY: undefined,
+        });
+    }
+
+    getQuoteButtonCoords = (quoteButtonPosition: string, rects: DOMRectList, spaceForMultilineSelection: number, startingSelectedElement: HTMLElement) => {
+        const sideBarRight = document.getElementById('sidebar-right');
+        const postCreateContainer = document.getElementsByClassName('post-create__container')[1].parentElement;
+        const sideBarRightHeader = document.getElementsByClassName('sidebar--right__header')[0];
+        const sidebarRightPostList = document.getElementsByClassName('post-list__dynamic--RHS')[0];
+        const spaceX = ((sideBarRight?.offsetLeft || 0) + 20) - rects[0].width;
+        const spaceY = (sideBarRightHeader.clientHeight + (postCreateContainer?.offsetTop || 0) + (sideBarRight?.offsetTop || 0)) - rects[0].height - sidebarRightPostList.scrollTop;
```

Comment:
do we really need this? It feels like the calculation based on the `window.getSelection` value should be sufficient, since the `getBoundingClientRect` already returns fixed values for the bounding box